### ISSUE TITLE
Call to protected method SimpleSAML\Composer\ModuleInstaller::getPackageBasePath()

### ISSUE THREAD

**Issue by mglaman** [simplesamlphp/composer-module-installer#20]
This is a weird error happening on 1.2.0 when the package gets removed

```
Call to protected method SimpleSAML\Composer\ModuleInstaller::getPackageBasePath() from scope SimpleSAML\Composer\ModuleInstallerPlugin_composer_tmp1 
```

trace

```
[2022-08-31T20:40:07.179Z] Exception trace:
[2022-08-31T20:40:07.179Z]   at phar:///usr/bin/composer/src/Composer/Plugin/PluginManager.php(275) : eval()'d code:60
[2022-08-31T20:40:07.179Z]  SimpleSAML\Composer\ModuleInstallerPlugin_composer_tmp1->uninstall() at phar:///usr/bin/composer/src/Composer/Plugin/PluginManager.php:435
[2022-08-31T20:40:07.179Z]  Composer\Plugin\PluginManager->uninstallPlugin() at phar:///usr/bin/composer/src/Composer/Plugin/PluginManager.php:348
[2022-08-31T20:40:07.179Z]  Composer\Plugin\PluginManager->uninstallPackage() at phar:///usr/bin/composer/src/Composer/Installer/PluginInstaller.php:115
[2022-08-31T20:40:07.179Z]  Composer\Installer\PluginInstaller->uninstall() at phar:///usr/bin/composer/src/Composer/Installer/InstallationManager.php:519
[2022-08-31T20:40:07.179Z]  Composer\Installer\InstallationManager->uninstall() at phar:///usr/bin/composer/src/Composer/Installer/InstallationManager.php:392
[2022-08-31T20:40:07.179Z]  Composer\Installer\InstallationManager->Composer\Installer\{closure}() at phar:///usr/bin/composer/vendor/react/promise/src/FulfilledPromise.php:28
[2022-08-31T20:40:07.180Z]  React\Promise\FulfilledPromise->then() at phar:///usr/bin/composer/src/Composer/Installer/InstallationManager.php:393
[2022-08-31T20:40:07.180Z]  Composer\Installer\InstallationManager->executeBatch() at phar:///usr/bin/composer/src/Composer/Installer/InstallationManager.php:334
[2022-08-31T20:40:07.180Z]  Composer\Installer\InstallationManager->downloadAndExecuteBatch() at phar:///usr/bin/composer/src/Composer/Installer/InstallationManager.php:242
[2022-08-31T20:40:07.180Z]  Composer\Installer\InstallationManager->execute() at phar:///usr/bin/composer/src/Composer/Installer.php:787
[2022-08-31T20:40:07.180Z]  Composer\Installer->doInstall() at phar:///usr/bin/composer/src/Composer/Installer.php:616
[2022-08-31T20:40:07.180Z]  Composer\Installer->doUpdate() at phar:///usr/bin/composer/src/Composer/Installer.php:281
[2022-08-31T20:40:07.180Z]  Composer\Installer->run() at phar:///usr/bin/composer/src/Composer/Command/RemoveCommand.php:298
[2022-08-31T20:40:07.180Z]  Composer\Command\RemoveCommand->execute() at phar:///usr/bin/composer/vendor/symfony/console/Command/Command.php:298
[2022-08-31T20:40:07.180Z]  Symfony\Component\Console\Command\Command->run() at phar:///usr/bin/composer/vendor/symfony/console/Application.php:1024
[2022-08-31T20:40:07.180Z]  Symfony\Component\Console\Application->doRunCommand() at phar:///usr/bin/composer/vendor/symfony/console/Application.php:299
[2022-08-31T20:40:07.180Z]  Symfony\Component\Console\Application->doRun() at phar:///usr/bin/composer/src/Composer/Console/Application.php:343
[2022-08-31T20:40:07.180Z]  Composer\Console\Application->doRun() at phar:///usr/bin/composer/vendor/symfony/console/Application.php:171
[2022-08-31T20:40:07.180Z]  Symfony\Component\Console\Application->run() at phar:///usr/bin/composer/src/Composer/Console/Application.php:138
[2022-08-31T20:40:07.180Z]  Composer\Console\Application->run() at phar:///usr/bin/composer/bin/composer:88
[2022-08-31T20:40:07.180Z]  require() at /usr/bin/composer:29
```

**Comment by tvdijen** [comment]
I tried to reproduce this by requiring a module and then remove it again, but it's working fine.. Can you run 'composer clearcache' and try again?

**Comment by mglaman** [comment]
I'm on PTO until Tuesday, but I'll try to reproduce. It was happening in our CI when removing a bunch of packages during automation. 

When we removed a package (https://www.drupal.org/project/simplesamlphp_auth) requiring the main package first, it just worked. It's weird. Like when doing an uninstall somehow the code for the plugin still ran even though it was removed? Which is why I imagine there is the temp in the trace.

So it might be more of a Composer bug.

Unfortunately I'm away from that computer and can't give more specifics until then.

**Comment by mglaman** [comment]
Sorry, I wasn't able to get a great way to reproduce. I'll just close this.

Basically, Package A required simplesamlphp. When we tried removing Package A along with numerous others, we got this crash. It's also an uncommon occurrence (we have a set of packages by default and then remove them to test a bump in another project's major version.) It's also probably not even a bug with this module, but something in Composer or who knows where, given the filename of `ModuleInstallerPlugin_composer_tmp1`
